addons/calendar/static/src/js/services/calendar_notification_service.js

```diff
@@ -0,0 +1,118 @@
+         */
+        function displayCalendarNotification(notifications) {
+            var lastNotifTimer = 0;
+
+            // Clear previously set timeouts and destroy currently displayed calendar notifications
+            browser.clearTimeout(getNextCalendarNotifTimeout);
+            Object.values(calendarNotifTimeouts).forEach((notif) => browser.clearTimeout(notif));
+            calendarNotifTimeouts = {};
+
+            // For each notification, set a timeout to display it
+            notifications.forEach(function (notif) {
+                var key = notif.event_id + ',' + notif.alarm_id;
+                if (key in calendarNotif) {
+                    return;
+                }
+                calendarNotifTimeouts[key] = browser.setTimeout(function () {
+                    var notificationID = env.services.notification.create(
+                        notif.message, {
+                            title: notif.title,
+                            type: "warning",
+                            sticky: true,
+                            onClose: () => {
+                                delete calendarNotif[key];
+                            },
+                            buttons: [{
+                                    name: env._t('OK'),
+                                    primary: true,
+                                    onClick: () => {
+                                        env.services.rpc('/calendar/notify_ack')
+                                            .then(env.services.notification.close(calendarNotif[key]));
+                                    },
+                                },
+                                {
+                                    name: env._t('Details'),
+                                    onClick: () => {
+                                        env.services.action.doAction('calendar.action_calendar_event_notify', {
+                                            resId: notif.event_id
+                                        }).then(env.services.notification.close(calendarNotif[key]));
+                                    },
+                                },
+                                {
+                                    name: env._t('Snooze'),
+                                    onClick: () => {
+                                        env.services.notification.close(calendarNotif[key]);
+                                    },
+                                },
+                            ],
+                        }
+                    );
+                    calendarNotif[key] = notificationID;
+                }, notif.timer * 1000);
+                lastNotifTimer = Math.max(lastNotifTimer, notif.timer);
+            });
+
+            // Set a timeout to get the next notifications when the last one has been displayed
+            if (lastNotifTimer > 0) {
+                getNextCalendarNotifTimeout = browser.setTimeout(getNextCalendarNotif, lastNotifTimer * 1000);
```

Comment:
not sure about the variable name: nextCalendarNotifTimeout instead ? This isn't a getter...